Quantized ConvTranspose2d

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35460 Quantized ConvTranspose1d
* **#35371 Quantized ConvTranspose2d**
* #35370 Base class for the quantized ConvTranspose

Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

Commit:
Quantized ConvTranspose2d

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

jerryzh168 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -563,3 +563,138 @@ def _output_padding(self, input, output_size, stride, padding, kernel_size):
+        assert weight_post_process.dtype == torch.qint8, \
+            'Weight observer must have a dtype of qint8'
+        qweight = _quantize_weight(mod.weight.float(), weight_post_process)
+        qconv = cls(mod.in_channels, mod.out_channels, mod.kernel_size,
```
I believe we have `output_padding` as well

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

jerryzh168 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -563,3 +563,157 @@ def _output_padding(self, input, output_size, stride, padding, kernel_size):
+
+        # TODO(z-a-f): Once the quantized `.flip` is implemented,
+        #              we can remove the requantization
+        qscale = w.q_scale()
```
wait, what if weight is per channel quantized?

z-a-f replied:
Not supported yet -- I will send a PR with the per-channel support later on

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

z-a-f commented:
Coming up: unittests

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

Commit:
Update on "Quantized ConvTranspose2d"


Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)
